esp-hal/src/dma/mod.rs

```diff
@@ -2034,6 +2036,14 @@ impl DmaTxBuf {
                     return Err(DmaBufError::InvalidAlignment);
                 }
             } else {
+                #[cfg(any(esp32,esp32s2))]
+                if buffer.len() % 4 != 0 && buffer.as_ptr() as usize % 4 != 0 {
```

Comment:
```suggestion
                if buffer.len() % 4 != 0 && buffer.as_ptr().is_aligned_to(4) {
```

Reply:
*cry*
```
error[E0658]: use of unstable library feature 'pointer_is_aligned_to'
    --> src/dma/mod.rs:2040:61
     |
2040 |                 if buffer.len() % 4 != 0 && buffer.as_ptr().is_aligned_to(4) {
     |                                                             ^^^^^^^^^^^^^
     |
     = note: see issue #96284 <https://github.com/rust-lang/rust/issues/96284> for more information
     = help: add `#![feature(pointer_is_aligned_to)]` to the crate attributes to enable
     = note: this compiler was built on 2024-09-03; consider upgrading it if it is out of date
```

Reply:
Damn sorry. `.cast::<u32>().is_aligned()` should be stable but at that point I'm not sure it's better than %4.